### ISSUE TITLE
Aggregations: redacting an event should also redact all of its edits.

### ISSUE THREAD

**Issue by lampholder** [matrix-org/synapse#5594]

**Comment by jryans** [comment]
Should fix https://github.com/vector-im/riot-web/issues/10111.

**Comment by anoadragon453** [comment]
I assume this should redact *any* aggregation on the event, like reactions, not just edits correct?

**Comment by anoadragon453** [comment]
Since this will create a few more redaction events, do you need them to be returned in the response to [/_matrix/client/r0/rooms/{roomId}/redact/{eventId}/{txnId}](https://matrix.org/docs/spec/client_server/unstable#put-matrix-client-r0-rooms-roomid-redact-eventid-txnid)?

**Comment by anoadragon453** [comment]
After meeting: We're just going to continue to return the original redaction event ID and leave the rest to come down /sync.

**Comment by anoadragon453** [comment]
For dealing with calling `/relations` for a redacted event, we're going to just return an empty `chunks` array.

**Comment by anoadragon453** [comment]
I believe we need to handle this for `/aggregations` as well.

**Comment by anoadragon453** [comment]
So this is currently half-implemented by blocking users from requesting relations/aggregations on a redacted event, but clients aren't told down sync that all edits should be redacted (and thus they have to work this out for themselves).

@lampholder Is the current state sufficient for Riot? Is it something that should be changed still, as in is it a good long-term solution or is it all a bit hacky from the client's side still?

**Comment by anoadragon453** [comment]
Reopening as edits are still not redacted.

**Comment by ShadowJonathan** [comment]
I don't think that synapse should be handling this, as relations can be arbitrary, and a redaction being a cascading event should be explicitly noted.

This sounds more like a spec issue, then, and/or it should be taken into the Relations MSC.

I think it's better for clients to manually find and redact all "appropriate" events, i.e. not reactions to their own events, but if they're removing a message, only all edits. This should be a case-by-case situation.
